engine/gamesys/src/gamesys/test/test_gamesys.h

```diff
@@ -626,7 +622,9 @@ void GamesysTest<T>::SetUp()
 
     assert(dmGameObject::RESULT_OK == dmGameSystem::RegisterComponentTypes(m_Factory, m_Register, m_RenderContext, &m_PhysicsContext, &m_ParticleFXContext, &m_SpriteContext,
                                                                                                     &m_CollectionProxyContext, &m_FactoryContext, &m_CollectionFactoryContext,
-                                                                                                    &m_ModelContext, &m_LabelContext, &m_TilemapContext, &m_SoundContext));
+                                                                                                    &m_ModelContext, &m_LabelContext, &m_TilemapContext));
+
+    dmGameObject::SortComponentTypes(m_Register);
```

Comment:
Just like the engine.cpp, once all types are registered, sort them on their priority, and they'll update in the expected order.